[refactor] each notification aborts itself if rabbitMQ is not enabled

peetucket opened the pull request:
## Why was this change made? 🤔

Fixes #3507 - make the notifications responsible for checking if rabbitMQ is enabled instead of the caller


## How was this change tested? 🤨

Added new tests

NOTE: The diffs in the specs are less substantial than they appear.  The existing tests were moved into a new context of "RabbitMQ enabled", with mocking of that setting to true.  Then a new context block was added mocking RabbitMQ to disabled to verify the messages are not sent.

peetucket commented:
> Can you remove from existing tests where no longer necessary? For example, `legacy_metadata_update_spec`?

That test is mocking the RabbitMQ setting to true and the call to publish so that it can verifying the call to publish... which still seems reasonable to me...if get rid of the mocking, we would need to get rid of the test expectaction too, e.g.

https://github.com/sul-dlss/dor-services-app/blob/main/spec/requests/legacy_metadata_update_spec.rb#L177

And in the other tests that do this as well.

justinlittman commented:
No, the test expectation would remain the same in that case -- you'd expect Notification::Updated to have been invoked.

peetucket commented:
> No, the test expectation would remain the same in that case -- you'd expect Notification::Updated to have been invoked.

ok, i just removed the rabbitmq setting mocking in those other tests (not needed anymore since it defaults to false) - everything else stays the same in the test
